### PR TITLE
Added radix_sort_3.c

### PR DESCRIPTION
**Pull request by debnath003**
#### Description of Change
Added readix_sort_3.c for the radix sort algorithm in DSA.

#### References
Issue #1316

#### Checklist

- [ ] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [ ] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Comment by debnath003** [comment]
@Panquesito7 @tjgurwara99 @alexpantyukhin could you all please approve my PR?

**Comment by tjgurwara99** [comment]
Please don't tag people directly unless you're stuck somewhere and it's very urgent. This does not qualify for urgent - people here are volunteering their time, so we will get to it when we get some time. Thanks.

**Comment by debnath003** [comment]
I have submitted my PR request @Panquesito7 @alexpantyukhin , awaiting review

**Comment by debnath003** [comment]
@Panquesito7  @tjgurwara99 @alexpantyukhin could you all please approve my PR? It has been a long time since I submitted my PR

**Comment by debnath003** [comment]
@Panquesito7 @tjgurwara99 @alexpantyukhin could you all please approve my PR? It has been a long time since I submitted my PR

File: sorting/radix_sort_3.c [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+    printf("Sorted array: ");
+    for (int i = 0; i < n; i++) {
+        printf("%d ", array.data[i]);
+    }
+    printf("\n");
+
+    // Free the dynamically allocated memory
+    free(array.data);
+
```
Comment (tjgurwara99) [inline sorting/radix_sort_3.c]: This needs to go inside its own `test` functions. Please read out [contribution guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#test) for guidance of what is expected of your PR.

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]

**Comment by debnath003** [comment]
@tjgurwara99 Rectified the code by given `test()` function, please review this!

**Pull request closed without merging**